CentralHub.Api/Threading/CancellableMutex.cs

```diff
@@ -0,0 +1,74 @@
+namespace CentralHub.Api.Threading;
+
+public sealed class CancellableMutex<T> : IDisposable
+{
+    private readonly SemaphoreSlim _semaphoreSlim = new SemaphoreSlim(1, 1);
+
+    private readonly T _innerObject;
+
+    public CancellableMutex(T innerObject)
```

Comment:
is the templating really needed? i've only seen it in this language

Reply:
ah i see, you need it for using lambda in the call to the lock right?

Reply:
Yes you get the value in the lambda and when you are in the lambda the lock is held